src/gui/widgets/Knob.cpp

```diff
@@ -380,8 +367,36 @@ bool Knob::updateAngle()
 void Knob::drawKnob( QPainter * _p )
 {
 	bool enabled = this->isEnabled();
-	QColor currentArcColor = enabled ? m_arcActiveColor : m_arcInactiveColor;
-	QColor currentLineColor = enabled ? m_lineActiveColor : m_lineInactiveColor;
+	QColor currentArcTopColor;
+	QColor currentArcBottomColor;
+	QColor currentLineColor;
+	QColor currentCenterTopColor;
+	QColor currentCenterBottomColor;
+
+	switch (knobNum())
+	{
+		case knobSmall_17: case knobBright_26: case knobDark_28:
+		{
+			currentArcTopColor = enabled ? m_arcTopActiveColor : m_arcTopInactiveColor;
+			currentArcBottomColor = enabled ? m_arcBottomActiveColor : m_arcBottomInactiveColor;
+			currentLineColor = enabled ? m_lineActiveColor : m_lineInactiveColor;
+			currentCenterTopColor = enabled ? m_centerTopActiveColor : m_centerTopInactiveColor;
+			currentCenterBottomColor = enabled ? m_centerBottomActiveColor : m_centerBottomInactiveColor;
+			break;
+		}
+		case knobVintage_32:
+		{
+			currentLineColor = QApplication::palette().color(QPalette::Active, QPalette::Shadow);
+			currentArcTopColor = QColor(QApplication::palette().color(QPalette::Active, QPalette::Shadow));
+			currentArcBottomColor = QColor(QApplication::palette().color(QPalette::Active, QPalette::Shadow));
+			currentArcBottomColor.setAlpha(70);
+			currentCenterTopColor = enabled ? m_centerTopActiveColor : m_centerTopInactiveColor;
+			currentCenterBottomColor = enabled ? m_centerBottomActiveColor : m_centerBottomInactiveColor;
```

Comment:
`currentCenterTopColor` and `currentCenterBottomColor` are the same between all knobs, we can probably pull this out and always set it after the switch.